yt/yt/server/queue_agent/queue_exporter.cpp

```diff
@@ -78,25 +79,55 @@ namespace {
 // Unix timestamp T corresponds to TInstant::Seconds(T).
 // Most of calculations are done with unix time, since we chose our granularity to be in seconds.
 
-//! Find export unix ts range containing #unixTs for a given #exportPeriod.
+//! Find export unix ts range containing #now for a given #exportConfig.
 //!
 //! Returns [#beginTs; endTs), where endTs is also export unix ts describing the range.
 //! Alternatively, beginTs is export unix ts that comes before endTs.
-std::pair<ui64, ui64> GetExportUnixTsRange(ui64 unixTs, TDuration exportPeriod)
+std::pair<ui64, ui64> GetExportUnixTsRange(
+    TInstant now,
+    const TQueueStaticExportConfig& exportConfig)
 {
-    auto period = exportPeriod.Seconds();
-    YT_VERIFY(period > 0);
+    if (exportConfig.ExportPeriod) {
+        auto period = exportConfig.ExportPeriod->Seconds();
+        YT_VERIFY(period > 0);
+
+        std::pair<ui64, ui64> result;
+        result.first = (now.Seconds() / period) * period;
+        result.second = result.first + period;
+        return result;
+    }
+
+    static const auto UTCTimezone = NDatetime::TTimeZone{};
+    auto nowCivilSecond = NDatetime::Convert(now, UTCTimezone);
+    TCronExpression cronExpression{*exportConfig.ExportCronSchedule};
+    return std::pair<ui64, ui64>{
+        NDatetime::Convert(cronExpression.CronPrev(nowCivilSecond), UTCTimezone).Seconds(),
+        NDatetime::Convert(cronExpression.CronNext(nowCivilSecond), UTCTimezone).Seconds()};
+}
 
-    std::pair<ui64, ui64> result;
-    result.first = (unixTs / period) * period;
-    result.second = result.first + period;
-    return result;
+std::pair<ui64, ui64> GetExportUnixTsRange(
+    ui64 unixTs,
+    const TQueueStaticExportConfig& exportConfig)
+{
+    return GetExportUnixTsRange(TInstant::Seconds(unixTs), exportConfig);
 }
 
 //! The greatest export unix ts lower than #now according to #exportPeriod.
-ui64 GetExportUnixTsUpperBound(TInstant now, TDuration exportPeriod)
+ui64 GetExportUnixTsUpperBound(
+    TInstant now,
+    const TQueueStaticExportConfig& exportConfig)
 {
-    return GetExportUnixTsRange(now.Seconds(), exportPeriod).first;
+    return GetExportUnixTsRange(now, exportConfig).first;
+}
+
+//! Get the difference between #now and the previous CRON
+//! occurence, according to the provided #cronExpressionString.
+ui64 GetCronExpressionPeriod(ui64 now, const TString& cronExpressionString) {
```

Comment:
Let's either rename this function (like SecondsSinceSmth) to show that it just subtracts now - last occurrence, or rename `now` -> something signifying that it is the unix ts of a current cron expression occurrence.